ament_cmake_core/cmake/core/ament_add_default_options.cmake

```diff
@@ -0,0 +1,42 @@
+# Copyright 2022 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#
+# Explicit call to add default CMake options (e.g. BUILD_SHARED_LIBS)
+#
+# .. note:: It can be called multiple times, but should be called
+#   once for any package that requires the options.
+#
+# :param EXCLUDE_BUILD_SHARED_LIBS: Exclude the BUILD_SHARED_LIBS option
+#
+# @public
+#
+macro(ament_add_default_options)
+  # TODO(CH3): Would be good to parse args to skip options next time.
+  set(aado_options EXCLUDE_BUILD_SHARED_LIBS)
+  set(aado_oneValueArgs)
+  set(aado_multiValueArgs)
```

Comment:
On a related note, since this is a macro, consider unsetting these variables at the end. So they don't end up in the scope later.